contract/contracts/Offer.sol

```diff
@@ -0,0 +1,288 @@
+        }
+
+        srcAsset.safeTransfer(receiver_, amount_);
+
+        emit AssetWithdrawn(msg.sender, amount_);
+    }
+
+    /**
+     * @notice Close the offer. Only the creator can close it
+     */
+    function close() external {
+        if (status != Status.Open) {
+            revert OfferError(ErrorType.OfferNotOpen);
+        }
+
+        if (msg.sender != creator) {
+            revert OfferError(ErrorType.NotCreator);
+        }
+
+        status = Status.Closed;
+
+        emit OfferClosed(msg.sender);
+    }
+
+    /**
+     * @notice Accept the offer.
+             Anybody can accept the offer with acceptAmount of destination asset.
+             Source asset will be sent to receiver
+     * @param receiver_ user address who will receive the source asset deposited
+     */
+    function accept(address receiver_) external {
+        if (status != Status.Open) {
+            revert OfferError(ErrorType.OfferNotOpen);
+        }
+
+        status = Status.Accepted;
+
+        destAsset.safeTransferFrom(msg.sender, address(this), acceptAmount);
+        srcAsset.safeTransfer(receiver_, totalDeposits);
+
+        emit OfferAccepted(msg.sender);
+    }
+
+    /**
+     * @notice Gets payment balance for domain owner
+     * @return balance payment balance
+     */
+    function paymentBalanceForDomainOwner()
+        public
+        view
+        returns (uint256 balance)
+    {
+        if (withdrawPayments[domainOwner]) {
+            return 0;
+        }
+
+        unchecked {
+            balance =
+                (acceptAmount * commissionRate) /
+                Config.COMMISSION_RATE_SCALE;
+        }
+    }
+
+    /**
+     * @notice Gets payment balance for depositor
+     * @param depositor_ depositor address
+     * @return balance payment balance
+     */
+    function paymentBalanceForDepositor(
+        address depositor_
+    ) public view returns (uint256 balance) {
+        if (withdrawPayments[depositor_]) {
+            return 0;
+        }
+
+        uint256 depositAmount = deposits[depositor_];
+
+        unchecked {
+            balance =
+                (acceptAmount * depositAmount) /
+                totalDeposits -
+                (acceptAmount * depositAmount * commissionRate) /
+                Config.COMMISSION_RATE_SCALE /
+                totalDeposits;
+        }
+    }
+
+    /**
+     * @notice Domain owner calls this function to withdraw payment
+     * @param receiver_ user address who will receive the payment
+     */
+    function withdrawPaymentForDomainOwner(address receiver_) external {
+        if (status != Status.Accepted) {
+            revert OfferError(ErrorType.OfferNotAccepted);
+        }
+
+        if (msg.sender != domainOwner) {
+            revert OfferError(ErrorType.NotDomainOwner);
+        }
+
+        uint256 amount = paymentBalanceForDomainOwner();
+
+        destAsset.safeTransfer(receiver_, amount);
+
+        withdrawPayments[msg.sender] = true;
```

Comment:
did not check withdrawPayments - so one can make repeated withdrawal

Reply:
Checked here https://github.com/polymorpher/otc-country/blob/otc-contract/contract/contracts/Offer.sol#L209-L223

Reply:
okay